compiler/rustc_hir_typeck/src/method/suggest.rs

```diff
@@ -348,6 +348,11 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             err.downgrade_to_delayed_bug();
         }
 
+        if tcx.ty_is_opaque_future(rcvr_ty) && item_name.name == sym::poll {
+            err.help(&format!("method `poll` found on `{ty_str}` for `Pin<&mut Self>`, see documentation for `std::pin::Pin`"));
```

Comment:
I don't think just saying "see documentation" or "look at the async book" is very helpful here, imo.

We should mention that the self type must be pinned or something more actionable here. Ideally we'd link to some example that actually shows what it means to use `Box::pin` or `pin_mut!`. Otherwise, someone trying to call `poll` here will be led not very much further to a solution to their issue.

Reply:
Maybe we should link to https://rust-lang.github.io/async-book/04_pinning/01_chapter.html#pinning-in-practice

Also "found on `{ty_str}` for `Pin<&mut Self>`" seems a bit roundabout. Maybe just say "found for `Pin<&mut {ty_str}>`"?